### PR TITLE
Add pluralise filter

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
Get the plural form for an item for a given number of items.

```njk
{{ 1 | pluralise("mouse") }}
{{ 2 | pluralise("house") }}
{{ 2 | pluralise("house", { number: false }) }}
{{ 2 | pluralise("mouse", { plural: "mice" }) }}
{{ 2 | pluralise("mouse", { plural: "mice", number: false }) }}
```

Output

```html
1 mouse
2 houses
houses
10 mice
mice
```

### CODE REVIEW

File: docs/filters/string.md [hunk 1]
```diff
@@ -97,6 +97,28 @@ Output
...
+{{ 2 | pluralise("mouse", { plural: "mice", number: false }) }}
+```
+
+Output
+
+```html
+1 mouse
+2 houses
+houses
+10 mice
```
Comment (fofr) [inline docs/filters/string.md:118]:
```suggestion
2 mice
```

File: lib/filters/string.js [hunk 2]
```diff
@@ -51,6 +51,30 @@ export function isString (value) {
...
+/**
+ * Get the plural form for an item for a given number of items.
+ *
+ * @see {@link http://blog.gnclmorais.com/simple-pluralise-in-javascript}
+ *
+ * @example
+ * pluralise(1, 'mouse') // 1 mouse
+ * pluralise(2, 'house') // 2 houses
+ * pluralise(2, 'house', { number: false }) // houses
+ * pluralise(2, 'mouse', { plural: 'mice' }) // 10 mice
```
Comment (fofr) [inline lib/filters/string.js:63]:
```suggestion
 * pluralise(2, 'mouse', { plural: 'mice' }) // 2 mice
```

File: lib/filters/string.js [hunk 3]
```diff
@@ -51,6 +51,30 @@ export function isString (value) {
...
+ * @param {string} [plural={}] - Plural form
+ * @param {boolean} [number={}] - Output number alongside word
+ * @return {string} Plural form for `count`
+ */
+export function pluralise (count, singular, { plural, number } = {})  {
+  const message = count === 1 ? singular : (plural || `${singular}s`);
+
+  return number === false ? message : `${count} ${message}`;
+}
+
```
Comment (fofr) [inline lib/filters/string.js:77]: It's worth pointing out that this is only valid for languages like English with two plural states (`one` and `other`), some languages have multiple words dependent on the number of things (eg `one`, `few`, `many`, `other` – see Russian and Polish as examples)

Comment (paulrobertlloyd) [reply lib/filters/string.js:78]: Good point; have added a note to the docs. Might be able to use `Intl` here in the future, but given most services use English, can defer anything more complex

**Pull request merged**